include/cvc5/cvc5_proof_rule.h

```diff
@@ -2266,9 +2266,332 @@ enum ENUM(ProofRule) : uint32_t
 };
 // clang-format on
 
+#ifdef CVC5_API_USE_C_ENUMS
+#undef EVALUE
+#define EVALUE(name) CVC5_PROOF_REWRITE_RULE_##name
+#endif
+
+/**
+ * This enumeration represents the rewrite rules used in a rewrite proof. Some
+ * of the rules are internal ad-hoc rewrites, while others are rewrites
+ * specified by the RARE DSL. This enumeration is used as the first argument to
+ * the :cpp:enumerator:`DSL_REWRITE <cvc5::ProofRule::DSL_REWRITE>` proof rule.
+ */
+enum ENUM(ProofRewriteRule) : uint32_t
+{
+  EVALUE(NONE),
+  // RARE rules
+  // ${rules}$
+  EVALUE(ARITH_PLUS_ZERO),
```

Comment:
This is somewhat minor, is it possible to generate a one line comment for each rule here?
Something like
/** Auto-generated from RARE rule arith-plus-zero */